libs/geoservices/covid/src/lib/pages/maps/components/lockdown-map/lockdown-map.component.ts

```diff
@@ -0,0 +1,258 @@
+  public countyBoxModel: BehaviorSubject<CountyStateBox> = new BehaviorSubject(undefined);
+  public ngOnInit(): void {
+    this.mapService.loaded.subscribe((map) => {
+      this.getStats();
+      const zoomThreshold = 3;
+
+      map.addSource('county-lines', {
+        type: 'vector',
+        url: 'mapbox://gsepulveda96.countylines'
+      });
+
+      map.addSource('state-lines', {
+        type: 'vector',
+        url: 'mapbox://gsepulveda96.statelines'
+      });
+
+      map.addLayer(
+        {
+          id: 'counties-lockdowns',
+          type: 'fill',
+          source: 'county-lines',
+          layout: {
+            visibility: 'none'
+          },
+          'source-layer': 'county-lines',
+          paint: {
+            'fill-outline-color': '#303030'
+          }
+        },
+        'road-motorway-trunk'
+      );
+
+      map.addLayer(
+        {
+          id: 'counties-claims',
+          type: 'fill',
+          source: 'county-lines',
+          'source-layer': 'county-lines',
+          paint: {
+            'fill-outline-color': '#FFFFFF'
+          }
+        },
+        'counties-lockdowns'
+      );
+
+      map.addLayer(
+        {
+          id: 'statelines',
+          type: 'line',
+          source: 'state-lines',
+          'source-layer': 'state-lines',
+          paint: {
+            'line-color': '#000000',
+            'line-width': 1
+          }
+        },
+        'state-label'
+      );
+
+      map.on('mousemove', 'counties-lockdowns', (e) => {
+        const feature = e.features[0];
+        this.countyBoxModel.next({
+          county: feature.properties.NAME
+        });
+      });
+
+      map.on('click', 'counties-lockdowns', (e) => {
+        const feature = e.features[0];
+        const hoveredCounty = this.statData.filter((county) => county.fips === feature.properties.fips);
+        const updatedDate: Date = hoveredCounty[0]['lockdownInfo'][0]
+          ? hoveredCounty[0]['lockdownInfo'][0]['updated']
+          : null;
+        const createdDate: Date = hoveredCounty[0]['lockdownInfo'][0]
+          ? hoveredCounty[0]['lockdownInfo'][0]['created']
+          : null;
+        this.infoBoxModel.next({
+          name: feature.properties.NAME,
+          state: hoveredCounty[0]['state'],
+          fips: hoveredCounty[0]['fips'],
+          fipsNum: hoveredCounty[0]['fipsNum'],
+          claims: hoveredCounty[0]['claims'],
+          sites: hoveredCounty[0]['sites'],
+          lockdowns: hoveredCounty[0]['lockdowns'],
+          lockdownInfo: hoveredCounty[0]['lockdownInfo'],
+          updated: updatedDate,
+          created: createdDate
+        });
+      });
+
+      map.dragRotate.disable();
+    });
+  }
+
+  public getFipsLockdownData(fipsCode: string) {
+    const lockdownUrl = 'https://nodes.geoservices.tamu.edu/api/covid/lockdowns/active/county/' + fipsCode;
+    const requests = forkJoin([this.http.get(lockdownUrl)]);
```

Comment:
Separation of concerns.

As you make heavier user of the http library, you've probably noticed you need to prepare the URL before sending it out the request. Should the host or endpoint change, it would require going into these components and making several updates.

Separate the http access into a service which can be injected into this component and consumed here. In this way, you abstract out the URL, error handling, and any data transformation/cleaning up, before it makes it to the component, on top of having to make updates in a single place.

There is already a data-access library for this covid project and there are several exported services for various resources relative to this project. The library scope is:

'@tamu-gisc/geoservices/data-access'

This particular resource call is already abstracted as `LockdownsService.getLockdownForCounty`.

Reply:
I noticed that I'm not using this function at the moment. I will remove it for now. If I need to re add it I will implement it using LockdownService